File: spring/src/main/java/io/github/wavesync/repository/WeaponMasterRepository.java
package io.github.wavesync.repository;
import io.github.wavesync.entity.WeaponMaster;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;
import java.util.Optional;


@Repository
public interface WeaponMasterRepository extends JpaRepository<WeaponMaster, Long> {

    WeaponMaster findByName(String name);

    @Query("""
        SELECT w
        FROM WeaponMaster w
        WHERE REPLACE(w.name, ' ', '') = :name
    """)
    Optional<WeaponMaster> findByNameWithoutSpaces(@Param("name") String name);
}

File: fastapi/app/repositories/weapon_master_repository.py
from typing import Optional

from sqlalchemy import func, select
from sqlalchemy.orm import Session

from app.models.weapon_master import WeaponMaster


def find_by_name(db: Session, name: str) -> Optional[WeaponMaster]:
    return db.scalar(select(WeaponMaster).where(WeaponMaster.name == name))


def find_by_name_without_spaces(db: Session, name: str) -> Optional[WeaponMaster]:
    return db.scalar(select(WeaponMaster).where(func.replace(WeaponMaster.name, " ", "") == name))
